src/components/flowchart/flowchart.js

```diff
@@ -503,14 +504,22 @@ export class FlowChart extends Component {
       this.handleSingleNodeClick(node);
 
       // the hold shift only happens on clicking a node first
-      if (event.shiftKey) {
+      // but only if no filters are currently applied.
+      if (event.shiftKey && !this.props.isSlicingPipelineApplied) {
```

Comment:
Is this to prevent slice on slice ? Do we in future expect to have a slice-on-slice feature (like multi-level-slicing) ? If so, how easy is it to extend this feature to multi-slice ?

Reply:
hey that's correct, that's to prevent to slice on a sliced pipeline. I think what you pointed out there is a very extreme edge case and don't think we would every want to slice on a slice? What we might do is after slicing a pipeline, user can register it as a new pipeline. Then we just reset the slicing state and start again

Reply:
On Viz side atleast, if we have an initialPipelineState, we can iteratively update the initialPipelineState = slicedState for recursive slicing. For now, I wanted to know if there is any idea of supporting this case in future :)

Reply:
good suggestion 👍 for now as this is meant to be MVP1 and we want to keep them simple, and we will need to look at user data to see whether we want to improve the feature or not. This can be something taking into account for future discussion